azure/scope/machine.go

```diff
@@ -375,7 +375,8 @@ func (m *MachineScope) VMExtensionSpecs() []azure.ResourceSpecGetter {
 		})
 	}
 
-	bootstrapExtensionSpec := azure.GetBootstrappingVMExtension(m.AzureMachine.Spec.OSDisk.OSType, m.CloudEnvironment(), m.Name())
+	cpuArchitectureType, _ := m.cache.VMSKU.GetCapability(resourceskus.CPUArchitectureType)
```

Comment:
Since `VMExtensionSpecs()` does not return an error, I don't think I can do much with the bool return type. The `azure.GetBootstrappingVMExtension` only checks if the `cpuArchitectureType` is arm64, and will set the extension's version to 1.1.1. Otherwise, we would default to 1.0, which would work even with an empty `cpuArchitectureType`.